connectors/sources/mssql.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Microsoft SQL source module is responsible to fetch documents from Microsoft SQL."""
+from sqlalchemy import create_engine
+from sqlalchemy.engine import URL
+
+from connectors.sources.generic_database import GenericBaseDataSource
+
+SECURED_CONNECTION = False
+# Below schemas are system schemas and the tables of the systems schema's will not get indexed
+SYSTEM_SCHEMA = [
+    "INFORMATION_SCHEMA",
+    "db_owner",
+    "db_accessadmin",
+    "db_securityadmin",
+    "db_ddladmin",
+    "db_backupoperator",
+    "db_datareader",
+    "db_datawriter",
+    "db_denydatareader",
+    "db_denydatawriter",
+    "sys",
+]
+
+QUERIES = {
+    "PING": "SELECT 1+1",
+    "ALL_TABLE": "SELECT table_name FROM information_schema.tables WHERE TABLE_SCHEMA = '{schema}'",
+    "TABLE_PRIMARY_KEY": "SELECT C.COLUMN_NAME FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS T JOIN INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE C ON C.CONSTRAINT_NAME=T.CONSTRAINT_NAME WHERE C.TABLE_NAME='{table}' and C.TABLE_SCHEMA='{schema}' and T.CONSTRAINT_TYPE='PRIMARY KEY' ",
+    "TABLE_DATA": 'SELECT * FROM {schema}."{table}"',
+    "TABLE_LAST_UPDATE_TIME": "SELECT last_user_update FROM sys.dm_db_index_usage_stats WHERE object_id=object_id('{schema}.{table}')",
+    "TABLE_DATA_COUNT": 'SELECT COUNT(*) FROM {schema}."{table}"',
+    "ALL_SCHEMAS": "SELECT s.name from sys.schemas s inner join sys.sysusers u on u.uid = s.principal_id",
+}
+
+
+class MSSQLDataSource(GenericBaseDataSource):
+    """Microsoft SQL Server"""
+
+    def __init__(self, configuration):
+        """Setup connection to the Microsoft SQL database-server configured by user
+
+        Args:
+            configuration (DataSourceConfiguration): Object of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.mssql_driver = self.configuration["mssql_driver"]
+        self.secured_connection = self.configuration["secured_connection"]
+        self.connection_string = ""
+        self.queries = QUERIES
+        self.dialect = "Microsoft SQL"
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for database-server configured by user
+
+        Returns:
+            dictionary: Default configuration
+        """
+        mssql_configuration = super().get_default_configuration().copy()
+        mssql_configuration.update(
+            {
+                "mssql_driver": {
+                    "value": "ODBC Driver 18 for SQL Server",
+                    "label": "Microsoft SQL Driver",
+                    "type": "str",
+                },
```

Comment:
Sure we can update it as  `Microsoft SQL Driver Name` and if we want to add an example it would become something like `Microsoft SQL Driver Name (ODBC Driver 18 for SQL Server)`.

Let us know your thoughts on what should we keep.

Reply:
I'm still not sure what you mean. 

What are the possible values for this field? What does it affect? How, as a user, I can understand what to input into this field?

Reply:
Hey @moxarth-elastic, any updates on this?

Reply:
Currently, we have labeled it as `Microsoft SQL Driver Name (ODBC Driver 18 for SQL Server)`. It means the user needs to pass the driver's name. By executing the command `odbcinst -j` user will able to find the path of the `odbcinst.ini` file. By opening `odbcinst.ini` file the user finds the installed driver name in their system. So, the user will find the name of the driver from `odbcinst.ini` file.   

Reference links: 
 - https://stackoverflow.com/questions/53575855/how-to-find-list-of-odbc-drivers-installed-on-linux
 - https://docs.sqlalchemy.org/en/14/dialects/mssql.html#hostname-connections